lighthouse-core/gather/gather-runner.js

```diff
@@ -187,17 +206,56 @@ class GatherRunner {
       const status = `Retrieving: ${gatherer.name}`;
       return chain.then(_ => {
         log.log('status', status);
-        return gatherer.afterPass(options, passData);
-      }).then(ret => {
+        const artifactPromise = Promise.resolve().then(_ => gatherer.afterPass(options, passData));
+        gathererResults[gatherer.name].push(artifactPromise);
+        return artifactPromise.catch(err => {
+          if (!err.recoverable) {
```

Comment:
yeah hard to tell at this point I suppose how frequently recoverable errors will be thrown from scratch or if they'll be caught and rethrown, delaying this decision for the next PR seems good anyway then :)